src/components/DonutChart.js

```diff
@@ -61,101 +67,209 @@ const DonutChart = React.createClass({
 
   getInitialState () {
     return {
-      activeIndex: -1
+      activeIndex: this.props.activeIndex || -1
     };
   },
 
+  componentWillMount () {
+    this._setupD3Functions(this.props);
+  },
+
   componentDidMount () {
-    this.setState({
-      activeIndex: this.props.activeIndex
-    });
+    this._animateChart();
   },
 
   componentWillReceiveProps (newProps) {
+    if (!_isEqual(this.props.data, newProps.data)) {
+      this._setupD3Functions(newProps);
+      this._animateChart();
+    }
+
     if (newProps.activeIndex !== this.props.activeIndex) {
       this.setState({
         activeIndex: newProps.activeIndex
       });
+
+      this._animateActiveArc(this.props.activeIndex, newProps.activeIndex);
+    }
+  },
+
+  shouldComponentUpdate (nextProps, nextState) {
+    return !_isEqual(this.props, nextProps) || !_isEqual(this.state, nextState);
+  },
+
+  _setupD3Functions (props) {
+    const dataSets = props.data.map(item => {
+      return item.value;
+    });
+    const valueTotal = dataSets.length ? dataSets.reduce((a, b) => {
+      return a + b;
+    }) : 0;
```

Comment:
I played around with reduce and you can accomplish the map and reduce in one step like so:
```
const valueTotal = props.data.reduce((previous, current) => {
  return { value: previous.value + current.value };
}).value;
```
Your way might be more readable? But I had fun poking around with reduce and wanted to share and see what you think.